marshmallow/fields.py

```diff
@@ -531,12 +531,26 @@ def _serialize(self, value, attr, obj):
         return [self.container._serialize(value, attr, obj)]
 
     def _deserialize(self, value, attr, data):
-        if utils.is_collection(value):
-            # Convert all instances in typed list to container type
-            return [self.container.deserialize(each) for each in value]
-        else:
+        if not utils.is_collection(value):
             self.fail('invalid')
 
+        result = []
+        errors = {}
+        for idx, each in enumerate(value):
+            try:
+                result.append(self.container.deserialize(each))
+            except ValidationError as e:
+                result.append(e.data)
+                item_errors = e.messages
+                if utils.is_collection(item_errors) and len(item_errors) == 1:
```

Comment:
I'm not sure this is necessary. For consistency, why not keep the errors in a list, even if there is only one error?